pylint/checkers/misc.py

```diff
@@ -23,20 +23,32 @@
 
 from pylint.interfaces import IRawChecker
 from pylint.checkers import BaseChecker
-from pylint.utils import OPTION_RGX
+from pylint.utils import OPTION_RGX, MessagesHandlerMixIn
 
 
-MSGS = {
-    'W0511': ('%s',
-              'fixme',
-              'Used when a warning note as FIXME or XXX is detected.'),
-    'W0512': ('Cannot decode using encoding "%s",'
-              ' unexpected byte at position %d',
-              'invalid-encoded-data',
-              'Used when a source line cannot be decoded using the specified '
-              'source file encoding.',
-              {'maxversion': (3, 0)}),
-}
+class ByIdDisabledMessagesChecker(BaseChecker):
+
+    """checks for messages that are disabled by id instead of symbol."""
+
+    __implements__ = IRawChecker
+
+    # configuration section name
+    name = 'miscellaneous'
+    msgs = {'I0023': ('%s',
+                      'by-id-disabled-message',
+                      'Used when a message is disabled by id.'),}
+
+    options = ()
+
+    def process_module(self, module):
+        """inspect the source file to find messages deactivated by id."""
+        for (mod_name, msg_id, msg_symbol, lineno) in (
+                MessagesHandlerMixIn.get_by_id_disabled_msgs()):
```

Comment:
This can be on the same line, it won't be that long. Or the `get_by_id_disabled_msgs()` could be on a separate line, with the results saved in a variable.